tests/foreman/api/test_syncplan.py

```diff
@@ -493,43 +494,102 @@ def test_positive_repeatedly_add_remove(self):
             self.assertEqual(len(syncplan.read().product), 0)
 
 
-@stubbed()
 class SyncPlanSynchronizeTestCase(APITestCase):
     """Tests specific to synchronizing sync plans."""
 
     @classmethod
     def setUpClass(cls):
-        """Create an organization and products which can be re-used in
-        tests."""
+        """Create an organization which can be re-used in tests."""
         super(SyncPlanSynchronizeTestCase, cls).setUpClass()
         cls.org = entities.Organization().create()
-        cls.products = [
-            entities.Product(organization=cls.org).create() for _ in range(2)
-        ]
 
-    @stubbed()
+    def validate_repo_content(
+            self, repo, content_types, after_sync=True, max_attempts=10):
+        """Check whether corresponding content is present in repository before
+        or after synchronization is performed
+
+        :param repo: Repository entity instance to be validated
+        :param content_types: List of repository content entities that
+            should be validated (e.g. package, erratum, puppet_module)
+        :param bool after_sync: Specify whether you perform validation before
+            synchronization procedure is happened or after
+        :param int max_attempts: That value is basically introduced for slow
+            systems when user sure that sync procedure can take more than 300
+            seconds (30*10 where 10 is a default value for max_attempts
+            variable)
+
+        """
+        for i in range(max_attempts):
+            try:
+                for content in content_types:
+                    if after_sync:
+                        self.assertGreater(repo.content_counts[content], 0)
+                    else:
+                        self.assertFalse(repo.content_counts[content])
+                break
+            except AssertionError:
+                sleep(30)
+                repo = repo.read()
+        else:
+                raise AssertionError(
+                    'Repository contains invalid number of content entities')
+
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_product(self):
-        """Create a sync plan with one product and sync it.
+        """Create a sync plan with one product and sync it automatically.
 
-        @Assert: A sync plan is created with one product and product can be
-        synchronized.
+        @Assert: Product is synchronized successfully.
 
         @Feature: SyncPlan
         """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
```

Comment:
As per my understanding sync plan works like this:
Say you create a sync plan with current time (say 3.08 PM) and put it in 'hourly'.  It tries to run the sync plan immediately and if there are no products, it ignores it.  The next run will be only at 4.08 PM.

I am surprised to hear that the sync plan runs automatically whenever you add a product to it, regardless of the sync plan scheduled time.  Can you confirm this? If this happens, this is a bug.